plugins/poll/assets/javascripts/discourse/components/poll-options.gjs

```diff
@@ -0,0 +1,98 @@
+import Component from "@glimmer/component";
+import { fn } from "@ember/helper";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import routeAction from "discourse/helpers/route-action";
+import dIcon from "discourse-common/helpers/d-icon";
+import i18n from "discourse-common/helpers/i18n";
+import PollOptionsDropdown from "./poll-options-dropdown";
+
+export default class PollOptionsComponent extends Component {
+  @service currentUser;
+
+  isChosen = (option) => {
+    return this.args.votes.includes(option.id);
+  };
+
+  get classes() {
+    return this.args.isIrv ? "irv-poll-options" : "";
+  }
+
+  @action
+  sendClick(option) {
+    this.args.sendOptionSelect(option);
+  }
+
+  @action
+  sendRank(option, rank = 0) {
+    this.args.sendOptionSelect(option, rank);
+  }
+  <template>
+    <ul class={{this.classes}}>
+      {{#each @options as |option|}}
+        {{#if @isIrv}}
+          <div
+            tabindex="0"
+            class="irv-poll-option"
+            data-poll-option-id={{option.id}}
+            data-poll-option-rank={{option.rank}}
+          >
+            {{#if this.currentUser}}
+              <PollOptionsDropdown
+                @rank={{option.rank}}
+                @option={{option}}
+                @irvDropdownContent={{@irvDropdownContent}}
+                @sendRank={{this.sendRank}}
+              />
+            {{else}}
+              <button
+                class="btn btn-default"
+                onclick={{routeAction "showLogin"}}
+              >{{i18n "poll.options.irv.login"}}</button>
+            {{/if}}
+            <span class="option-text">{{option.html}}</span>
+          </div>
+        {{else}}
+          <li tabindex="0" data-poll-option-id={{option.id}}>
+            {{#if this.currentUser}}
+              <button {{on "click" (fn this.sendClick option)}}>
```

Comment:
DButton

Reply:
@jjaffeux this sends the formatting off completely.  What's the benefit of using DButton in this case and in general?

Reply:
Just having homogeneous code, but for places where this is too much of a hassle you can forget it

Reply:
the issue with DButton here is that it adds hover behaviour we don't want or need in this circumstance which we would then have to explicitly back out in CSS.  I think it may make sense to avoid that in this special case?

![image](https://github.com/discourse/discourse/assets/35533304/4150263a-be75-45c8-8f72-9998b18495bd)

Reply:
I've updated one of the buttons here to DButton (login button for IRV for anon) as it makes overall sense.  Very happy to revisit if a concern.